data_preprocessing/name_preprocessing.py

```diff
@@ -0,0 +1,230 @@
+    people_db.create_positions_csv()
+    people_db.merge_duplicates()
+
+    people_db.store_to_disk(out_file)
+    print("Merging names took", time.time() - initial_time)
+
+
+
+def parse_doc_metadata_csv(csv_path, people_db=None):
+    """
+    Parses the metadata for each document in a csv of tobacco document metadata
+
+    For each document belonging to a certain company,
+    finds Person that that document author alias maps to,
+    and adds that company to the position counter of that Person
+
+     Goes through documents and updates authors/recipients' positions
+     based on organization associated with document
+
+     Adds people to db_current if their alias is associated with a doc but is not in
+     db_current yet
+
+     Ignores:
+         organizations not in RAW_ORG_TO_CLEAN_ORG_DICT
+         docs with more than 1 organization
+         docs with more than 4 recipients
+         docs with no people associated with them
+
+
+    :param csv_path: Path
+    :param db_to_add: PeopleDatabase
+    :return:
+    """
+
+    if not people_db:
+        people_db = PeopleDatabase()
+
+    alias_to_person_dict = people_db.get_alias_to_person_dict()
+
+    # get lists of dicts for authors and recipients. each of them has 3 fields:
+    # general, person, organization
+    people_dicts = get_au_and_rc_by_document(csv_path, return_type='both')
+
+    for person in people_dicts:
+
+        orgs = set()
+        for org in person['organization']:
+            if org not in RAW_ORG_TO_CLEAN_ORG_DICT:
+                continue
+            orgs.add(RAW_ORG_TO_CLEAN_ORG_DICT[org])
+
+        if len(orgs) != 1:
+            continue
+        org = list(orgs)[0]
+        aliases = person['person']
+        aliases.extend(person['general'])
+        if not aliases or len(aliases) >= 4:
+            continue
+
+        for alias in aliases:
+            if alias not in alias_to_person_dict:
+                people_db.add_person_raw(alias)
+            else:
+                alias_to_person_dict[alias].positions[org] += 1
+
+
+def load_documents_to_dataframe(path) -> pd.DataFrame:
+    """
+    Loads the Dunn documents into a pandas Dataframe
+    :return: pd.DataFrame
+    """
+    df = pd.read_csv(path).fillna('')  # pylint: disable=C0103
+    return df
+
+
+def get_au_and_rc_by_document(path, return_type='both') -> list:
+    """
+    Creates a list of documents such that each element consists of a dict with keys
+
+    returns either just information about the authors, the recipients, or both
+
+    Note(SR): to give authors and recipients the same structure, I have renamed 'au' and 'rc'
+    to general.
+
+    'au', 'au_org', 'au_person' OR 'rc', 'rc_org', 'rc_person'.
+    These keys map to info about the authors/recipients and organizations associated with docs
+    :param path: Path
+    :param return_type: str
+    :return: list
+    """
+
+    if return_type not in ['authors', 'recipients', 'both']:
```

Comment:
I modified this function such that it returns a list of dicts. 
Depending on the return_type, that list will have all of the info about authors, recipients, or both

For each dict, 'au' and 'rc' are stored as general, 'au_person' and 'rc_person' as person etc.